strawberry/field.py

```diff
@@ -279,6 +317,22 @@ def field(
     default: Any = UNSET,
     default_factory: Union[Callable, object] = UNSET,
 ) -> StrawberryField:
+    ...
+
+
+def field(
+    resolver: Optional[_RESOLVER_TYPE] = None,
+    *,
+    name: Optional[str] = None,
+    is_subscription: bool = False,
+    description: Optional[str] = None,
+    init: Optional[bool] = False,
```

Comment:
this is a workaround for making pyright think that a field with a resolver is not added to the constructor signature (see the test below)